framework/tester.py

```diff
@@ -0,0 +1,163 @@
+        rtype = server['response']
+        if rtype == 'static':
+            content = fill_template(server['response_content'])
+            srv = deproxy_server.StaticDeproxyServer(port=port,
+                                                     response=content)
+        else:
+            raise Exception("Invalid response type: %s" % str(rtype))
+
+        self.__deproxy_manager.add_server(srv)
+        return srv
+
+    def __create_backend(self, server):
+        srv = None
+        sid = server['id']
+        if server['type'] == 'nginx':
+            srv = self.__create_srv_nginx(server, sid)
+        elif server['type'] == 'deproxy':
+            srv = self.__create_srv_deproxy(server)
+        self.__servers[sid] = srv
+
+    def __create_servers(self):
+        for server in self.backends:
+            self.__create_backend(server)
+
+    def get_server(self, sid):
+        """ Return client with specified id """
+        if not self.__servers.has_key(sid):
+            return None
+        return self.__servers[sid]
+
+    def get_servers_id(self):
+        """ Return list of registered servers id """
+        return self.__servers.keys()
+
+    def __create_clients(self):
+        for client in self.clients:
+            self.__create_client(client)
+
+    def get_client(self, cid):
+        """ Return client with specified id """
+        if not self.__clients.has_key(cid):
+            return None
+        return self.__clients[cid]
+    
+    def get_clients_id(self):
+        """ Return list of registered clients id """
+        return self.__clients.keys()
+
+    def get_tempesta(self):
+        """ Return Tempesta instance """
+        return self.__tempesta
+
+    def __create_tempesta(self):
+        config = ""
+        if self.tempesta.has_key('config'):
+            config = self.tempesta['config']
+        self.__tempesta = control.Tempesta()
+        self.__tempesta.config.set_defconfig(fill_template(config))
+
+    def start_all_servers(self):
+        for id in self.__servers:
+            srv = self.__servers[id]
+            srv.start()
+            if srv.state != stateful.STATE_STARTED:
+                raise Exception("Can not start server %s" % id)
+
+    def start_tempesta(self):
+        self.__tempesta.start()
+        if self.__tempesta.state != stateful.STATE_STARTED:
+            raise Exception("Can not start Tempesta")
+
+    def start_all_clients(self):
+        for id in self.__clients:
+            client = self.__clients[id]
+            client.start()
+            if client.state != stateful.STATE_STARTED:
+                raise Exception("Can not start client %s" % id)
+
+    def setUp(self):
+        tf_cfg.dbg(3, '\tInit test case...')
+        self.__create_servers()
+        self.__create_tempesta()
+        self.__create_clients()
+        self.__deproxy_manager.start()
+        # preventing race between manager start and servers start
+        time.sleep(0.2)
+
+    def tearDown(self):
```

Comment:
Are the methods, `setUp()` and `tearDown()` called by some third-party library? If so, then this must be commented (who and when calls them).

Reply:
they are called from `unittest.TestCase`